src/Microsoft.AspNetCore.Sockets.Client/LongPollingTransport.cs

```diff
@@ -114,6 +118,27 @@ public Task StartAsync(Uri url, IChannelConnection<Message> application)
             }
         }
 
+        private IEnumerable<Message> ReadMessages(ReadOnlySpan<byte> payload)
+        {
+            if (payload.Length == 0)
```

Comment:
It looks like when poll times out, we get an `OperationCancelledException` and the whole transport shuts down...